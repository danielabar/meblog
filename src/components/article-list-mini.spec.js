import React from "react"
import { render, screen } from "@testing-library/react"
import ArticleListMini from "./article-list-mini"

describe("ArticleListMini", () => {
  const articles = [
    {
      node: {
        id: "1",
        title: "Article 1",
        published_at: "2024-04-10",
        slug: "/article-1",
      },
    },
    {
      node: {
        id: "2",
        title: "Article 2",
        published_at: "2024-04-09",
        slug: "/article-2",
      },
    },
  ]

  it("renders correctly", () => {
    const container = render(
      <ArticleListMini articles={articles} title="Recent Posts" />
    )
    expect(container).toMatchSnapshot()

    expect(screen.getByText("Recent Posts")).toBeInTheDocument()

    const articleTitles = screen.getAllByRole("link", { name: /article/i })
    expect(articleTitles.length).toBe(2)

    expect(screen.getByText("Article 1")).toBeInTheDocument()
    expect(screen.getByText("Article 2")).toBeInTheDocument()
  })
})
