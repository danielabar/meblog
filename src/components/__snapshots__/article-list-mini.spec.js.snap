// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleListMini renders correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        data-testid="article-list-mini"
      >
        <h2>
          Recent Posts
        </h2>
        <a
          href="/article-1"
        >
          <div>
            <span>
              Article 1
            </span>
            <div>
              2024-04-10
            </div>
          </div>
        </a>
        <a
          href="/article-2"
        >
          <div>
            <span>
              Article 2
            </span>
            <div>
              2024-04-09
            </div>
          </div>
        </a>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      data-testid="article-list-mini"
    >
      <h2>
        Recent Posts
      </h2>
      <a
        href="/article-1"
      >
        <div>
          <span>
            Article 1
          </span>
          <div>
            2024-04-10
          </div>
        </div>
      </a>
      <a
        href="/article-2"
      >
        <div>
          <span>
            Article 2
          </span>
          <div>
            2024-04-09
          </div>
        </div>
      </a>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
